packages/x-components/src/x-installer/x-installer/x-installer.ts

```diff
@@ -128,38 +133,54 @@ export class XInstaller {
   }
 
   /**
-   * Receives the {@link SnippetConfig | snippet config}, installs the plugin and initializes
-   * the Vue application.
+   * Retrieves the {@link SnippetConfig | snippet config} it is defined in the window.initX.
    *
-   * @param snippetConfig - The {@link SnippetConfig} that receives from snippet
-   * integration.
+   * @returns The snippet config if it is defined. Else undefined.
    *
-   * @returns Object with the {@link XAPI}, the {@link XBus}, the {@link XPlugin} and the Vue App
-   * used in the application.
+   * @internal
+   */
+  private retrieveSnippetConfig(): SnippetConfig | undefined {
+    if (typeof window.initX === 'function') {
+      return window.initX();
+    } else if (typeof window.initX === 'object') {
+      return window.initX;
+    }
+  }
+
+  /**
+   * Receives the {@link SnippetConfig | snippet config} or retrieves it from window.initX and
+   * installs the plugin and initializes the Vue application.
+   *
+   * @param snippetConfig - The {@link SnippetConfig} that receives from snippet integration.
+   *
+   * @returns If {@link SnippetConfig | snippet config} is passed or configured in window.initX,
+   * returns an object with the {@link XAPI}, the {@link XBus}, the {@link XPlugin} and the Vue App
+   * used in the application. Else, a rejected promise is returned.
    *
    * @public
    */
-  async init(snippetConfig: SnippetConfig): Promise<{
-    app: Vue | undefined;
-    api: XAPI | undefined;
-    bus: XBus;
-    plugin: PluginObject<XPluginOptions>;
-  }> {
-    const adapterConfig = this.getAdapterConfig(snippetConfig);
-    this.applyConfigToAdapter(adapterConfig);
-    const bus = this.createBus();
-    const pluginOptions = this.getPluginOptions();
-    const plugin = this.installPlugin(pluginOptions, bus);
-    const extraPlugins = await this.installExtraPlugins(snippetConfig, bus);
-    const app = this.createApp(extraPlugins, snippetConfig);
-    this.api?.setBus(bus);
+  async init(snippetConfig?: SnippetConfig): Promise<InitWrapper | void> {
+    const finalSnippetConfig = snippetConfig ?? this.retrieveSnippetConfig();
 
-    return {
-      api: this.api,
-      app,
-      bus,
-      plugin
-    };
+    if (finalSnippetConfig) {
+      const adapterConfig = this.getAdapterConfig(finalSnippetConfig);
+      this.applyConfigToAdapter(adapterConfig);
+      const bus = this.createBus();
+      const pluginOptions = this.getPluginOptions();
+      const plugin = this.installPlugin(pluginOptions, bus);
+      const extraPlugins = await this.installExtraPlugins(finalSnippetConfig, bus);
+      const app = this.createApp(extraPlugins, finalSnippetConfig);
+      this.api?.setBus(bus);
+
+      return {
+        api: this.api,
+        app,
+        bus,
+        plugin
+      };
+    }
+
+    return Promise.resolve();
```

Comment:
Shouldn't be better `return Promise.reject()` in this case? I'm thinking that basically we failed to install anthing.

Reply:
If this was a normal method that was doing only 1 thing yes, but as we have to support the 2 ways of loading x-components, and we don't want to stop in every client to ask them how they prefer to load, I would just return the Promise resolve, even if looks a little weird.